webapp/src/components/expanded_view/call_settings.tsx

```diff
@@ -0,0 +1,477 @@
+import type {
+    AudioDevices,
+} from 'src/types/types';
+import {getCallsClient} from 'src/utils';
+import styled from 'styled-components';
+
+import ControlsButton from './controls_button';
+
+type AudioDevicesListProps = {
+    deviceType: string;
+    devices: MediaDeviceInfo[];
+    currentDevice: MediaDeviceInfo | null;
+    onDeviceClick: (device: MediaDeviceInfo) => void;
+}
+
+const AudioDevicesList = ({deviceType, devices, currentDevice, onDeviceClick}: AudioDevicesListProps) => {
+    const {formatMessage} = useIntl();
+
+    // Note: this is system default, not the concept of default that we save in local storage in client.ts
+    const makeDeviceLabel = (device: MediaDeviceInfo) => {
+        if (device.deviceId.startsWith('default') && !device.label.startsWith('Default')) {
+            return formatMessage({defaultMessage: 'Default - {deviceLabel}'}, {deviceLabel: device.label});
+        }
+        return device.label;
+    };
+
+    const list = devices.map((device) => {
+        const isCurrentDevice = device.deviceId === currentDevice?.deviceId;
+        return (
+            <li
+                className='MenuItem'
+                key={`audio-${deviceType}-device-${device.deviceId}`}
+            >
+                <AudioDeviceButton
+                    className='style--none'
+                    onClick={() => onDeviceClick(device)}
+                    $isCurrentDevice={isCurrentDevice}
+                >
+                    <AudioDeviceName>
+                        {makeDeviceLabel(device)}
+                    </AudioDeviceName>
+                    { isCurrentDevice &&
+                    <AudioDeviceSelectedIcon>
+                        <TickIcon/>
+                    </AudioDeviceSelectedIcon>
+                    }
+                </AudioDeviceButton>
+            </li>
+        );
+    });
+
+    return (
+        <div className='Menu'>
+            <DevicesList
+                id={`calls-popout-audio-${deviceType}s-menu`}
+                className='Menu__content dropdown-menu'
+            >
+                {list}
+            </DevicesList>
+        </div>
+    );
+};
+
+const AudioDeviceSelectedIcon = styled.div`
+&&&& {
+    svg {
+      width: 14px;
+      height: 14px;
+      fill: var(--button-bg);
+    }
+}
+`;
+
+const AudioDeviceName = styled.span`
+  color: rgba(var(--center-channel-color-rgb), 0.56);
+  font-size: 14px;
+  width: 100%;
+  text-overflow: ellipsis;
+  overflow: hidden;
+`;
+
+const AudioDeviceButton = styled.button<{$isCurrentDevice: boolean}>`
+&&& {
+  display: flex;
+  background: ${({$isCurrentDevice}) => $isCurrentDevice ? 'rgba(28, 88, 217, 0.08)' : ''};
+  line-height: 20px;
+  padding: 8px 20px;
+  align-items: center;
+  gap: 8px;
+}
+`;
+
+const DevicesList = styled.ul`
+&&&& {
+  top: 0;
+  left: calc(100% + 4px);
+  overflow: auto;
+  width: 280px;
+  max-height: calc(100% + 36px);
+  border-radius: 8px;
+}
```

Comment:
nit: Is there a way to tell your formatter to indent the styled-components css? Also is it possible to set it to 4-spaces? If you really prefer 2, I can set mine to 2 then. I just always thought 4 fits the rest of the file better, but that's the only reason I would prefer it.

Reply:
We need the linter. Can't rely on people's IDEs :p

Reply:
Remember https://mattermost.atlassian.net/browse/MM-48266?